lua/cfc_weapon_lockouts/server/sv_locker.lua

```diff
@@ -0,0 +1,216 @@
+CFCWeaponLockouts = CFCWeaponLockouts or {}
+
+local lockWarns = {}
+
+function CFCWeaponLockouts.lockWeapon( ply, wep, lostWeapon )
+    if not IsValid( ply ) then return end
+
+    if not wep then --The caller only knows the weapon, and not the player
+        wep = ply
+        ply = wep.weaponLockoutOwner
+
+        if not IsValid( ply ) then return end
+    end
+
+    if not IsValid( wep ) or not wep:IsWeapon() or not ply:Alive() or CFCWeaponLockouts.NOT_LOCKABLE[wep:GetClass()] then return end
+
+    ply.weaponLockouts = ply.weaponLockouts or {}
+    local lockouts = ply.weaponLockouts
+    local weaponClass = wep:GetClass()
+
+    --If the caller thinks the weapon was lost, or if they are unsure and it was lost
+    if lostWeapon or ( lostWeapon == nil and not ply:HasWeapon( weaponClass ) ) then
```

Comment:
Can you bring
```lua
lostWeapon == nil and not ply:HasWeapon( weaponClass )
```
out into its own named variable so this check is easier to read?